rest_framework/utils/formatting.py

```diff
@@ -65,3 +65,30 @@ def markup_description(description):
         description = escape(description).replace('\n', '<br />')
         description = '<p>' + description + '</p>'
     return mark_safe(description)
+
+
+class lazy_format:
+    """
+    Delay formatting until it's actually needed.
+
+    Useful when the format string or one of the arguments is lazy.
+
+    Not using Django's lazy because it is too slow.
+    """
+    __slots__ = ('format_string', 'args', 'kwargs', 'result')
+
+    def __init__(self, format_string, *args, **kwargs):
+        self.result = None
+        self.format_string = format_string
+        self.args = args
+        self.kwargs = kwargs
+
+    def __str__(self):
+        if self.result is not None:
+            self.format_string, self.args, self.kwargs = None, None, None
+            return self.result
+        self.result = self.format_string.format(*self.args, **self.kwargs)
+        return self.result
```

Comment:
The following reads a little more clearly to me. Thoughts?
```python
if self.result is None:
    self.result = self.format_string.format(*self.args, **self.kwargs)
    self.format_string, self.args, self.kwargs = None, None, None
return self.result

```

Reply:
Yep, that's better. I applied it.